src/core/io/processinterface.cpp

```diff
@@ -55,7 +56,13 @@ bool ProcessInterface::initialise(bool paIsInput, CEventChainExecutionThread *co
   deinitialise();
 
   // Register interface
-  if(!(mIsListening = IOMapper::getInstance().registerObserver(getFullQualifiedApplicationInstanceName('.'), this))) {
+  if (PARAMS().length()) {  // If PARAMS is not empty we use the string specified here "as is" for the IO mapping to the configuration. Otherwise use the full qualified instance name of our FB.
+    if (!(mIsListening = IOMapper::getInstance().registerObserver(std::string(PARAMS().c_str()), this))) {
```

Comment:
please use `getStorage()` instead this removes the need to convert the `std::string` to a `char *` and then create a new `std::string` from it.

Reply:
Even better, use the explicit std::string cast operator